crates/sui-types/src/object.rs

```diff
@@ -329,11 +329,11 @@ impl MoveObject {
     pub fn get_total_sui(&self, resolver: &impl GetModule) -> Result<u64, SuiError> {
         let layout = self.get_layout(ObjectFormatOptions::with_types(), resolver)?;
         let move_struct = self.to_move_struct(&layout)?;
-        Ok(Self::get_total_sui_(&move_struct, 0))
+        Ok(Self::get_total_sui_in_struct(&move_struct, 0))
```

Comment:
am I understanding it correctly that a top level object cannot be a vector?

Reply:
Correct. A top-level object must always have `key`, and there's a verifier pass that checks any object with `key` has a `id: UID` field.